app/controllers/leads_controller.rb

```diff
@@ -9,22 +9,34 @@ class LeadsController < ApplicationController
   def index
     respond_to do |format|
       format.html {
-        @recent_objects = Lead.recent_from_project_id(sessions_current_project_id).where('parent_id is null').order(updated_at: :desc).limit(10)
+        one_week_ago = Time.now.utc.to_date - 7
```

Comment:
@kleintom you have a native relation returned from `roots_with_data`.  You should be able to chain addition .where, limit, order conditions directly on to that rather than dropping to Ruby?

Reply:
Yup, thanks.